PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningHallWidget.kt

```diff
@@ -0,0 +1,93 @@
+package com.pennapps.labs.pennmobile
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.widget.RemoteViews
+import com.google.gson.GsonBuilder
+import com.google.gson.reflect.TypeToken
+import com.pennapps.labs.pennmobile.adapters.DiningHallWidgetAdapter
+import com.pennapps.labs.pennmobile.api.DiningRequest
+import com.pennapps.labs.pennmobile.api.Serializer
+import com.pennapps.labs.pennmobile.classes.DiningHall
+import com.pennapps.labs.pennmobile.classes.Venue
+import com.squareup.okhttp.OkHttpClient
+import retrofit.RestAdapter
+import retrofit.client.OkClient
+import retrofit.converter.GsonConverter
+import java.util.concurrent.TimeUnit
+
+/**
+ * Implementation of App Widget functionality.
+ */
+class DiningHallWidget : AppWidgetProvider() {
```

Comment:
I would like to see more comments and spacing. Would be nice to document what you're doing.